src/wallet/api/wallet2_api.h

```diff
@@ -1144,6 +1154,519 @@ struct Wallet
+    */
+    virtual std::uint32_t adjustPriority(std::uint32_t priority) const = 0;
+    // QUESTION : Any suggestion for the function description below?
+    /**
+    * brief: coldTxAuxImport -
+    * param: ptx -
+    * param: tx_device_aux -
+    * note: sets status error on fail
+    */
+    virtual void coldTxAuxImport(const PendingTransaction &ptx, const std::vector<std::string> &tx_device_aux) const = 0;
+// TODO : wallet2::signed_tx_set & cryptonote::address_parse_info
+    /**
+    * brief: coldSignTx -
+    * param: ptx -
+    * param: exported_txs -
+    * param: dsts_info -
+    * param: tx_device_aux -
+    */
+//    virtual void coldSignTx(const PendingTransaction &ptx, signed_tx_set &exported_txs, std::vector<cryptonote::address_parse_info> &dsts_info, std::vector<std::string> & tx_device_aux) const = 0;
+    // TODO : this already exists, remove after checking
+    /**
+    * brief: useForkRules - check if the wallet uses the rules for hard fork version at current block height - early_blocks
+    * param: version - hard fork version
+    * param: early_blocks -
+    * return: true if fork rules are used
+    * note: sets status error on fail
+    */
+//    virtual bool useForkRules(std::uint8_t version, std::int64_t early_blocks) const = 0;
+// TODO : wallet2::transfer_details
+    /**
+    * brief: getTransferDetails -
+    * param: idx - index in transfer storage
+    * return: transfer details for given index
+    * note: sets status error on fail
+    */
+//    virtual const wallet2::transfer_details &getTransferDetails(std::size_t idx) const = 0;
+    /**
+    * brief: discardUnmixableOutputs - freeze all unmixable outputs
+    * note: sets status error on fail
+    */
+    virtual void discardUnmixableOutputs() = 0;
+    /**
+    * brief: setTxKey - set the transaction key (r) for a given <txid> in case the tx was made by some other device or 3rd party wallet
+    * param: txid -
+    * param: tx_key - secret transaction key r
+    * param: additional_tx_keys -
+    * param: single_destination_subaddress -
+    * note: sets status error on fail
+    */
+    // QUESTION : There were some cases already where I wished to have boost::optional in here. I have no explicit example, but in general e.g. for functions that return a bool, but have to return early on error, it would be cleaner to return boost::none instead of true/false, because the API user could forget to check the error status and may use an incorrect value. Similar for uint*_t, where we can't use e.g. -1 on error.
+    //            If we add access to boost::optional revisit all new functions to see which would benefit from it.
+    virtual void setTxKey(const std::string &txid, const std::string &tx_key, const std::vector<std::string> &additional_tx_keys, const boost::optional<std::string> &single_destination_subaddress) = 0;
+    /**
+    * brief: getDaemonAddress -
+    * return: daemon address
+    */
+    virtual std::string getDaemonAddress() const = 0;
```

Comment:
Is any info about the currently active daemon, the daemon the wallet is connected to, a candidate for `WalletState`?